### ISSUE TITLE
'--read_length' and '--macs_gsize' Setting Confusion

### ISSUE THREAD

**Issue by anijudy** [nf-core/atacseq#303]
### Description of the bug

Hi everyone!

I'm trying to run the pipeline according to the Usage Docs. It seems that the minimal sufficient command to run the pipeline is the following:
`nextflow run nf-core/atacseq --input samplesheet.csv --outdir <OUTDIR> --genome GRCh37 -profile docker`

Since I don't have dockers installed on my cluster, I used singularity instead. I also specified the version, 2.0. However, when I ran the command, the following error occurred.
```
nextflow run nf-core/atacseq \
         -resume \
         -profile singularity \
         -r 2.0 \
         --input design_both_runs.csv \
         --genome GRCh37 \
         --outdir results \
         --max_memory 600.GB \
         --max_cpus 24
```

```
Both '--read_length' and '--macs_gsize' not specified! Please specify either to infer MACS2 genome size for peak calling.
```

I checked the parameter documentation. If I use GRCh37, I shouldn't worry about setting up the 'macs_gsize' parameter, and therefore, there is no need to set up read_length either. I'm not sure whether this is a pipeline bug or not, but at least it's confusing to me. Can anyone help me make the pipeline work? Any help is appreciated!

Thanks,
Yifan

### Command used and terminal output

```console
Command:

nextflow run nf-core/atacseq \
         -resume \
         -profile singularity \
         -r 2.0 \
         --input design_both_runs.csv \
         --genome GRCh37 \
         --outdir results \
         --max_memory 600.GB \
         --max_cpus 24

Output:

NOTE: Nextflow is not tested with Java 1.8.0_60 -- It's recommended the use of version 11 up to 18

N E X T F L O W  ~  version 22.04.0
Launching `https://github.com/nf-core/atacseq` [small_sammet] DSL2 - revision: 0add18866a [2.0]


------------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
  nf-core/atacseq v2.0
------------------------------------------------------
Core Nextflow options
  revision       : 2.0
  runName        : small_sammet
  containerEngine: singularity
  launchDir      : /proj/CGEHcore/Yifan_testproject_ATACseq/preprocessing_new
  workDir        : /proj/CGEHcore/Yifan_testproject_ATACseq/preprocessing_new/work
  projectDir     : /Fingerlin/home/zhaoy/.nextflow/assets/nf-core/atacseq
  userName       : zhaoy
  profile        : singularity
  configFiles    : /Fingerlin/home/zhaoy/.nextflow/assets/nf-core/atacseq/nextflow.config

Input/output options
  input          : design_both_runs.csv
  outdir         : results

Reference genome options
  genome         : GRCh37
  fasta          : s3://ngi-igenomes/igenomes/Homo_sapiens/Ensembl/GRCh37/Sequence/WholeGenomeFasta/genome.fa
  gtf            : s3://ngi-igenomes/igenomes/Homo_sapiens/Ensembl/GRCh37/Annotation/Genes/genes.gtf
  bwa_index      : s3://ngi-igenomes/igenomes/Homo_sapiens/Ensembl/GRCh37/Sequence/BWAIndex/version0.6.0/
  bowtie2_index  : s3://ngi-igenomes/igenomes/Homo_sapiens/Ensembl/GRCh37/Sequence/Bowtie2Index/
  star_index     : s3://ngi-igenomes/igenomes/Homo_sapiens/Ensembl/GRCh37/Sequence/STARIndex/
  blacklist      : /Fingerlin/home/zhaoy/.nextflow/assets/nf-core/atacseq/assets/blacklists/v1.0/GRCh37-blacklist.v1.bed
  mito_name      : MT

Max job request options
  max_cpus       : 24
  max_memory     : 600.GB

!! Only displaying parameters that differ from the pipeline defaults !!
------------------------------------------------------
If you use nf-core/atacseq for your analysis please cite:

* The pipeline
  https://doi.org/10.5281/zenodo.2634132

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://github.com/nf-core/atacseq/blob/master/CITATIONS.md
------------------------------------------------------
WARN: =============================================================================
  --macs_gsize parameter has not been provided.
  It will be auto-calculated by 'khmer unique-kmers.py' using the '--read_length' parameter.
  Explicitly provide '--macs_gsize macs2_genome_size' to change this behaviour.
===================================================================================
Both '--read_length' and '--macs_gsize' not specified! Please specify either to infer MACS2 genome size for peak calling.
```

### System information

nextflow version 22.04.0.5698
nf-core/atacseq 2.0
singularity version 3.8.6

**Comment by JoseEspinosa** [comment]
Hi @anijudy, thanks for reporting.
We updated this feature for 2.0. The way it works now, as the message suggests, is that you should either set the value of the `--macs_gsize` parameter or otherwise provide the length of your reads with `--read_length`. The reason is that we provide a macs gsize for each of the read lengths of a given genome entry see [here](https://github.com/nf-core/atacseq/blob/0add18866a5f095e93b8ec345a12b76d675f4af9/conf/igenomes.config#L25-L31) for `GRCh37`. The values were pre-computed using the approach described [here](https://deeptools.readthedocs.io/en/develop/content/feature/effectiveGenomeSize.html#effective-genome-size). So in your case if your reads have a length of say 50, you will be able to make the pipeline work by just adding to your command `--read_length 50`. We will try to improve the docs before the next release.

**Comment by anijudy** [comment]
> Hi @anijudy, thanks for reporting. We updated this feature for 2.0. The way it works now, as the message suggests, is that you should either set the value of the `--macs_gsize` parameter or otherwise provide the length of your reads with `--read_length`. The reason is that we provide a macs gsize for each of the read lengths of a given genome entry see [here](https://github.com/nf-core/atacseq/blob/0add18866a5f095e93b8ec345a12b76d675f4af9/conf/igenomes.config#L25-L31) for `GRCh37`. The values were pre-computed using the approach described [here](https://deeptools.readthedocs.io/en/develop/content/feature/effectiveGenomeSize.html#effective-genome-size). So in your case if your reads have a length of say 50, you will be able to make the pipeline work by just adding to your command `--read_length 50`. We will try to improve the docs before the next release.

Dear @JoseEspinosa,

Thank you for your detailed response, which provided me with valuable insights. It appears that the requirement for specifying the read length, even when a genome is specified, is due to the exclusion of multimapping reads or the application of MAPQ filters. Upon closer examination of the table, I noticed that the values set for macs_gsize do not match those in the table. For example, in the case of read_length 50 and GRCh37, the macs_gsize is indicated as 2684219875, whereas the table shows it as 2685511504. I'm curious if there is a conversion or adjustment needed between the numbers in the table and those in the configuration file.

Thank you once again for your help!

Best regards,
Yifan

**Comment by JoseEspinosa** [comment]
Yes, we detected this, but as we run ourselves the program we thought it was better to keep the values that we obtained, see []here](https://github.com/nf-core/chipseq/pull/283)

**Comment by JoseEspinosa** [comment]
Available in v2.1.0
